core/templates/pages/contributor-dashboard-page/modal-templates/question-suggestion-review.directive.html

```diff
@@ -10,7 +10,7 @@ <h4 class="author-header">Submitted by <[authorName]></h4>
 
 <div class="modal-body">
   <section class="oppia-suggestion-review-container">
-    <div class="oppia-question-suggestion">
+    <div id="questionSuggestionDiv" class="oppia-question-suggestion">
```

Comment:
Usually CSS IDs are discouraged as it's difficult to guarantee they are unique within a large app. Can we use a class here to select this component?

Reply:
It's actually the other way round. Per the[ Specificity Rules for CSS](https://www.w3schools.com/css/css_specificity.asp), IDs are more specific than classes. Classes can be shared across a group of elements, whereas IDs cannot. 

That being true, one can validate the uniqueness of an ID before confirming one in a large app by performing a universal search in Visual Code. The needful has been done and the uniqueness of the ID is validated. 

Also, I tried checking if there is any getElementByClass() function called which refers to class (that being more specific than an ID, per your and @kevintab95 perspective), but could not find such an instance. All our codebase refers to getElementByID(). So have let it be.

Reply:
Yes, IDs are more specific than classes, but generally we want to avoid using IDs since it will lead to styling clashes and they cannot be re-used for styling purposes. But since we're only using it here for selection, it should be fine.

Let's make the name a bit more specific though and drop the "div"—"questionSuggestionReview"?

Reply:
Please avoid `id`s. Our templates are reused in the codebase and our pages are dynamic. So doing a static search in the codebase is not sufficient to ensure that there aren't duplicates on a page. What happens if this template is used multiple times on the same page? Please prefer classnames.

Reply:
The template pertains to a modal, and the ID is only for its modal content. If the template is used anywhere else, the same behavior would be expected (sticky with an increased scrollable view). While I cannot perceive why a page would call the same modal thrice, I get your point of, in general, not sticking to strict specificity when referring to elements. Thus, I have modified the code to reflect classnames.

Reply:
Thanks! Just to clarify my point regarding reusability, templates can be modularised as well and used independent of the context in which it is displayed. So, a "modal" template could contain multiple moving parts and those can be used in a non-modal context.